highway.tex

```diff
@@ -24,807 +26,1323 @@
 
-We have left to show that an honest validator will only send messages satisfying Limited Naivety Criterion. This is because any message that clears $V$'s verification buffer either:
+\desc{
+
+\begin{center}
+    {\bf Round Exponent Maintenance Strategy for $V\in \cV$} 
+\end{center}
+% \damian{I wrote a slightly simpler version than what was written here before. I don't think it matters much how this is done. Feel free to roll it back to the previous version. }
+\begin{enumerate}
+\item At tick $i=0$, initialize $n_v(j)= \nmin$ for all $j$. Initialize $\cntsucc=0$.
+\item At tick $i>0$,  let $m:=n_v(i)$. Whenever $i$ is divisible by $2^{m+1}$:
+% \item At tick $i>0$,  let $m:=n_v(i)$. Whenever $i$ is divisible by $2^{m+1}$ and $n_v$ did not change in $C$ previous rounds (i.e., it was constant in the interval $[i-C2^{m+1}, i-1]$):
 \begin{enumerate}
-\item Was initially put into the verification buffer, but was (or was cited by) a vote that was later endorsed.
-\item Was sent by a validator who was not at the time in $E_V$.
+        \item Let $b_{\mathrm{fin}}$ denote the number of blocks finalized with confidence $t_0$ during the last $C$ rounds,
+        \item if $b_{\mathrm{fin}}<C_{\mathrm{fail}}$ then set $n_v(j)=\min(m+1, \nmax)$, for all $j\geq i$,
+        \item if $b_{\mathrm{fin}}>C_{\mathrm{succ}}$ then increment $\cntsucc=\cntsucc+1$, otherwise set $\cntsucc=0$.
+        \item If $i$ is divisible by $C2^{m+1}$ and $\cntsucc\geq D$, set $n_v(j)=\max(m-1, \nmin)$, for all $j\geq i$. Set $\cntsucc=0$.
+\end{enumerate}
 \end{enumerate}
-We note that the former type of votes are never naively cited by messages from $V$ and thus do not affect Limited Naivety Criterion. We claim additionally that votes of the latter type in $V$'s justifications contain no equivocations. This is because if $V$ receives two equivocating votes, say, $\lambda_1$ and $\lambda_2$, then by the time $V$ receives the second of these votes, the sender will be put in $E_V$ and the message will end up in the verification buffer. This shows that all of $V$'s messages under this rule satisfy Limited Naivety Criterion.
+}
 
 
-%\subsection{Impatience}
+In practice $t_0$ should probably be much lower than the confidence threshold used for "safety", for instance $t_0\approx 0.01$.
+%
+Even if in a single round, the block proposed by the leader gets finalized with the lower confidence $t_0$, if there are enough honest validators this confidence will increase in the future, as the summit grows in height. The constants $C_{\mathrm{fail}}, C_{\mathrm{succ}}, C, D $ are chosen so as to make sure that the "correct" round exponent is learned by the validators based on the "finality rate" feedback.
```

Comment:
And crucially, to prevent the summit from growing, someone would need to equivocate; and everyone can only _effectively_ equivocate once.

Reply:
I find this statement problematic. It may happen that there are summits S1, S2, S3 of respective heights 1,2, and 3, and some validator V may be crucial to all of them (it created unit of every level in every summit). If this validator later equivocate, none of this summits will grow any further (or - they will grow, but in a narrower version). And while it still would count as one _effective_ equivocation, it kind of breaks 3 rounds. I'll leave this paragraph as it is for now.

Reply:
You're right… what I meant to say was that: If you equivocate, eventually everyone will have you marked as faulty and you won't be able to break existing summits in future rounds after that anymore.